browser/ui/split_view/split_view_controller.cc

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/split_view/split_view_controller.h"
+
+#include "brave/browser/ui/split_view/split_view_view.h"
+#include "brave/browser/ui/split_view/split_view_web_panel_data.h"
+#include "brave/browser/ui/tabs/split_view_browser_data.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+
+SplitViewController::SplitViewController(TabStripModel* tab_strip_model)
+    : split_view_tab_tile_data_(new SplitViewBrowserData(tab_strip_model)),
+      split_view_web_panel_data_(new SplitViewWebPanelData()),
+      tab_strip_model_(*tab_strip_model) {
+  split_view_tab_tile_observation_.Observe(split_view_tab_tile_data_.get());
+}
+
+SplitViewController::~SplitViewController() = default;
+
+void SplitViewController::OnTileTabs(const TabTile& tile) {
+  if (IsActiveWebContentsIncludedIn(tile) && view_) {
```

Comment:
I know I lack some knowledge around the whole split view thing, but when is this the case that these events take place, but it is not a relevant WebContents.